x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateFormatter.java

```diff
@@ -59,19 +59,28 @@ class DateFormatter {
             new Builder().pattern("%s").javaPattern("ss").build(),
             new Builder().pattern("%T").javaPattern("HH:mm:ss").build(),
             new Builder().pattern("%U")
-                .javaFormat(t -> String.format(Locale.ROOT, "%02d", t.get(WeekFields.of(DayOfWeek.SUNDAY, 7).weekOfYear())))
+                .javaFormat(t -> String.format(Locale.ENGLISH, "%02d", t.get(WeekFields.of(DayOfWeek.SUNDAY, 7).weekOfYear())))
                 .build(),
-            new Builder().pattern("%u").javaFormat(t -> String.format(Locale.ROOT, "%02d", t.get(WeekFields.ISO.weekOfYear()))).build(),
+            new Builder().pattern("%u")
+                .javaFormat(t -> String.format(Locale.ENGLISH, "%02d", t.get(WeekFields.of(DayOfWeek.MONDAY, 4).weekOfYear())))
+                .build(),
+
             new Builder().pattern("%V")
-                .javaFormat(t -> String.format(Locale.ROOT, "%02d", t.get(WeekFields.of(DayOfWeek.SUNDAY, 7).weekOfWeekBasedYear())))
+                .javaFormat(t -> String.format(Locale.ENGLISH, "%02d", t.get(WeekFields.of(DayOfWeek.SUNDAY, 7).weekOfWeekBasedYear())))
+                .build(),
+            new Builder().pattern("%v")
+                .javaFormat(t -> String.format(Locale.ENGLISH, "%02d", t.get(WeekFields.of(DayOfWeek.MONDAY, 4).weekOfWeekBasedYear())))
```

Comment:
This is exactly how [MySQL spec](https://dev.mysql.com/doc/refman/8.4/en/date-and-time-functions.html#function_date-format) defines `%v`